### PR TITLE
Support progress meter

### PR DESCRIPTION
**Pull request by xukai92**
- Add progress meter (https://github.com/TuringLang/AdvancedHMC.jl/issues/57)
  - Also add flag to turn on or off the progress meter. This allows AHMC to run in silent mode, which is very necessary for large scale parallel use of the sampler. FYI, a friend of mine has been using AHMC on a quite large dataset. He runs the sampler in parallel using 40 cores for > 2 hours, and so far it seems to be all good. He also checked the inference results to that he obtained from his VAE implemented and found they are very consistent.
- Remove old intermediate `@info` for adaptation
- Merged `sample()` for adapted and non-adaptive `sample()`

### CODE REVIEW

File: src/sampler.jl [hunk 1]
```diff
@@ -37,40 +37,43 @@ sample(
     τ::AbstractProposal,
     θ::AbstractVector{T},
     n_samples::Int,
-    adaptor::Adaptation.AbstractAdaptor,
+    adaptor::Union{Nothing,Adaptation.AbstractAdaptor}=nothing,
```
Comment (yebai) [inline src/sampler.jl]: Can we define a special `NoAdaptation <: AbstractAdaptor ` instead of using `nothing`?

File: src/sampler.jl [hunk 2]
```diff
@@ -37,40 +37,43 @@ sample(
...
     n_adapts::Int=min(div(n_samples, 10), 1_000);
-    verbose::Bool=true
-) where {T<:Real} = sample(GLOBAL_RNG, h, τ, θ, n_samples, adaptor, n_adapts; verbose=verbose)
+    verbose::Bool=true,
+    progress::Bool=false
+) where {T<:Real} = sample(GLOBAL_RNG, h, τ, θ, n_samples, adaptor, n_adapts; verbose=verbose, progress=progress)
 
 function sample(
```
Comment (yebai) [inline src/sampler.jl]: Nice to unify `sample` interfaces.

File: src/sampler.jl (hunk 1 above)
Comment (xukai92) [reply src/sampler.jl]: Added in https://github.com/TuringLang/AdvancedHMC.jl/pull/65/commits/a0e1a9f770b38b7bee57a10aab17be950c065ea6

**Comment by yebai** [comment]
Looks nice, many thanks! @xukai92

**Pull request merged**